merkletree/pad_test.go

```diff
@@ -235,3 +239,190 @@ func TestPoliciesChange(t *testing.T) {
 		t.Error(key3, "value mismatch")
 	}
 }
+
+func TestTB(t *testing.T) {
+	key1 := "key"
+	val1 := []byte("value")
+
+	pad, err := NewPAD(NewPolicies(3, vrfPrivKey1), signKey, 3)
+	if err != nil {
+		t.Fatal(err)
+	}
+	tb, err := pad.TB(key1, val1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pk, ok := pad.signKey.Public()
+	if !ok {
+		t.Fatal("Couldn't retrieve public-key.")
+	}
+	tbb := tb.Serialize(pad.latestSTR.Signature)
+	if !pk.Verify(tbb, tb.Signature) {
+		t.Fatal("Couldn't validate signature")
+	}
+	// create next epoch and see if the TB is inserted as promised:
+	pad.Update(nil)
+
+	ap, err := pad.Lookup(key1)
```

Comment:
Should we also verify the proof of inclusion here, just to make this test as a sample for complete TB verification?